plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/HeteroDescribableConfigurator.java

```diff
@@ -133,8 +137,47 @@ public CNode describe(T instance, ConfigurationContext context) {
                 .getOrElseThrow(() -> new IllegalStateException("No configurator implementation to manage " + klazz));
     }
 
+    /**
+     * Matches suitable descriptors for the target.
+     * The fetch is trivial when the target implements a root {@link Describable} object.
+     * If not, we iterate to parent classes until we find a class which can provide the descriptor list in {@link Jenkins#getDescriptorList(Class)}.
+     * Then we go through all the descriptors and fined ones compliant with the target.
+     * @return Stream of descriptors which match the target
+     */
     private Stream<Descriptor<T>> getDescriptors() {
-        return Stream.ofAll(Jenkins.getInstance().getDescriptorList(target));
+        DescriptorExtensionList<T, Descriptor<T>> descriptorList = Jenkins.getInstance().getDescriptorList(target);
+        if (!descriptorList.isEmpty()) { // fast fetch for root objects
+            return Stream.ofAll(descriptorList);
+        }
+
+        DescriptorExtensionList parentDescriptorClassList = descriptorList;
+        Class<?> effectiveTarget = target;
+        while (parentDescriptorClassList.isEmpty() && effectiveTarget != null) {
+            final Class<Describable> match;
+            try {
+                match = (Class<Describable>)effectiveTarget;
+            } catch (Exception ex) {
+                break;
```

Comment:
is it worth logging this at all?

Reply:
I doubt. This is just a type check (cast and exception handling is cheaper than the reflection magic for generics)

Reply:
It could be good to have it in `DEBUG` so that someone having issues with that could create a custom logger for this and see the problem.. it's harmless in normal condition and can help debuging without any code modification.

Reply:
If it's not a bug, but a way to check the type and it's expected not to be the type, I don't like having a log for it. Maybe you can use _instanceof_ or even create a method to do that in this line and the other one.